website/blog/2022-12-13-pointer-events-in-react-native.md

```diff
@@ -0,0 +1,175 @@
+
+### Looking to web
+
+Web is a platform with similar challenges in scaling to many platforms while also considering future-proof design. The World Wide Web consortium (W3C) is tasked with setting standards and proposals to build a Web that is interoperable amongst different platforms and browsers.
+
+Most relevant for our needs, the W3C has defined behavior for an abstract form of input, called a pointer. The [Pointer Events](https://www.w3.org/TR/pointerevents3/) specification builds on mouse events and aims to provide a single set of events and interfaces for cross-device pointer input while still allowing for device-specific handling when necessary.
+
+Following the Pointer Events specification provides React Native users many benefits. Beyond addressing the problems mentioned earlier, it raises the capabilities of platforms that haven’t historically had to consider multi-input type interactions. Think attaching a bluetooth mouse to your Android phone or the Apple pencil supporting hover on the iPad M2.
+
+Being spec-complaint also provides opportunity for knowledge sharing between Web and React Native. Education of Web expectations around Pointer Events can doubly serve React Native developers. However, we also recognize that React Native requirements are different than web and our approach to specifications is best effort with well documented deviations so expectations are clear. There is related work of aligning certain Web standards to [reduce API fragmentation](https://github.com/react-native-community/discussions-and-proposals/pull/496) in accessibility and performance APIs.
+
+## Porting Web Platform Tests
+
+While the Pointer Events specification provides interfaces and behavior descriptions of the API, we found it wasn’t specific enough for us to confidently make changes and point to the specification as verification. However, web browsers use another mechanism to ensure compliance and interoperability — the [Web Platform Tests](https://web-platform-tests.org/)!
+
+The Web Platform Tests are written to work against the browser’s imperative DOM APIs — unsupported by React Native as it uses its own view primatives. This means that we aren’t able to code-share the tests with browsers and instead have an analogous testing API for React Native that makes it easier to port those Web Platform Tests.
+
+We implemented a new manual testing framework which we are now using for verifying our implementations through RNTester. These tests are tentatively named the RNTester Platform Tests and are still fairly basic. Our implementation provides an API to construct test cases as components themselves which are rendered and where the results are reported solely through the UI.
+
+![GIF showing a side by side comparison of the “Pointer Events hoverable pointer attributes test” running in React Native (iOS) on the left, and Web (the original implementation) on the right.](/blog/assets/pointer-events-wpt-demo.gif)
```

Comment:
To ensure that description are visible, you can use `<figure>` and `<figcaption>` elements instead an usual image syntax for Markdown, example:
* https://github.com/facebook/react-native-website/blob/035054cbdba34b7d1ff4eaed15af463e23431e61/docs/pressable.md?plain=1#L34:L39

This applies to all images used in this blog post.

Reply:
I believe these were written primarily as alt text to be consumed by assistive technologies.